File: pi/led/pi.py
import time
import RPi.GPIO as GPIO

# RPi.GPIO Layout verwenden (wie Pin-Nummern)
GPIO.setmode(GPIO.BOARD)

# Pin 18 (GPIO 24) auf Input setzen
GPIO.setup(18, GPIO.IN)

# Pin 11 (GPIO 17) auf Output setzen
GPIO.setup(11, GPIO.OUT)

# Dauersschleife
while 1:
  # LED immer ausmachen
  GPIO.output(11, GPIO.LOW)

  # GPIO lesen
  if GPIO.input(18) == GPIO.HIGH:
    # LED an
    GPIO.output(11, GPIO.HIGH)

    # Warte 100 ms
    time.sleep(0.1)

    # LED aus
    GPIO.output(11, GPIO.LOW)

    # Warte 100 ms
    time.sleep(0.1)

File: pi/led/pi.c
// WiringPi-Api einbinden
#include <wiringPi.h>

// C-Standardbibliothek einbinden
#include <stdio.h>

int main() {

  // Starte die WiringPi-Api (wichtig)
  if (wiringPiSetup() == -1)
    return 1;

  // Schalte GPIO 17 (=WiringPi Pin 0) auf Ausgang
  pinMode(0, OUTPUT);

  // Schalte GPIO 24 (=WiringPi Pin 5) auf Eingang
  pinMode(5, INPUT);

  // Dauerschleife
  while(1) {
    // LED immer ausmachen
    digitalWrite(0, 0);

    // GPIO lesen
    if(digitalRead(5)==1) {
      // LED an
      digitalWrite(0, 1);

      // Warte 100 ms
      delay(100);

      // LED aus
      digitalWrite(0, 0);

      // Warte 100 ms
      delay(100);
    }
  }
}
